src/gamma.jl

```diff
@@ -740,13 +740,21 @@ end
 
 Euler integral of the first kind ``\\operatorname{B}(x,y) = \\Gamma(x)\\Gamma(y)/\\Gamma(x+y)``.
 """
-function beta(a::Number, b::Number)
+beta(a::Number, b::Number) = _beta(map(float, promote(a, b))...)
+
+# here `T` is a floating point type but we don't want to restrict the implementation
+# to `AbstractFloat` or `Float64`
+function _beta(a::T, b::T) where {T<:Number}
```

Comment:
How do you know that it's a floating point type then? :-) It could also be a complex.

Reply:
I guess I was not completely precise here. What I meant was that the types are the result of calling `float` (this is what I called slightly incorrectly "floating point type"), so also in the case of complex numbers it has to be `Complex{T}` where `T` is a floating point number.